### DISCUSSION TITLE
[question] 请问G6的同级别的combo，如何才能做到不重叠呢？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [antvis/G6#6684] [Q&A]
请问：[G6的combo](https://g6.antv.antgroup.com/zh/examples/element/combo/#rect) 是否可以配置或怎么实现不重叠呢？
![image](https://github.com/user-attachments/assets/897fcdd8-b315-471c-b1c8-ebd5c83b06c3)

是否只能通过自定义layout才能实现？

**Comment by petercat-assistant** [comment]
在 G6 中，避免同级别的 combo 重叠通常需要通过自定义布局来实现。G6 提供了一些内置的布局算法，但如果这些算法不能满足需求，您可以通过编写自定义布局来控制 combo 的位置。

以下是一些可能的解决方案：

1. **使用内置布局**：尝试使用 G6 提供的内置布局，如 `dagre`、`circular` 等，这些布局可能会自动处理节点和 combo 的重叠问题。

2. **自定义布局**：如果内置布局不能满足需求，可以编写自定义布局算法。在自定义布局中，您可以手动设置 combo 的位置，确保它们不重叠。

3. **调整 combo 大小和间距**：通过调整 combo 的大小和间距，也可以减少重叠的可能性。

4. **使用力导向布局**：力导向布局可以通过设置合适的参数来减少重叠。

如果您需要进一步的帮助或示例代码，请告诉我！我可以为您提供更多详细信息。

**Reply by liaodalin19903** [reply]
dagre 会重叠。
